www/js/goals.js

```diff
@@ -196,13 +224,21 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
         $scope.goals.splice($scope.goals.indexOf(goal), 1);
     };
 
+    $scope.viewChallenge = function(challenge) {
+        $scope.challengeToView = challenge;
+        $scope.challengeInfoArray = [];
+        if($scope.profile.challenges.includes(challenge.id))
+            challengeLeaderboard(challenge.id);
+        openChallengeDetail();
+    }
+
     $scope.theUser = {};
     $scope.signup = function(){
         console.log($scope.theUser.username);
         var regConfig = {'username': $scope.theUser.username};
         console.log(regConfig);
         $ionicLoading.show({
-            template: '<ion-spinner icon="bubbles" class="costume"></ion-spinner>'
```

Comment:
Yes, the costume class is not in the style, the loading screen is the same without this code.